mpl: openbsd does not know about AI_V4MAPPED

Building on OpenBSD-7 and I get a compile-time error:

```
gmake[1]: Entering directory '/home/robl/src/mpich/build/src/mpl'
  CC       src/sock/mpl_sockaddr.lo
../../../src/mpl/src/sock/mpl_sockaddr.c:101:24: error: use of undeclared identifier 'AI_V4MAPPED'
    ai_hint.ai_flags = AI_V4MAPPED;
                       ^
1 error generated.
```

and it's true there is no mention of AI_V4MAPPED in the openbsd man page

--- a/src/mpl/src/sock/mpl_sockaddr.c
+++ b/src/mpl/src/sock/mpl_sockaddr.c
@@ -98,7 +98,6 @@ int MPL_get_sockaddr(const char *s_hostname, MPL_sockaddr_t * p_addr)
     ai_hint.ai_family = af_type;
     ai_hint.ai_socktype = SOCK_STREAM;
     ai_hint.ai_protocol = IPPROTO_TCP;
-    ai_hint.ai_flags = AI_V4MAPPED;
     ret = getaddrinfo(s_hostname, NULL, &ai_hint, &ai_list);
     if (ret) {
         return ret;

